src/serialization/tokenizer.hpp

```diff
@@ -20,45 +20,90 @@
 
 #include "buffered_istream.hpp"
 
+#include <array>
 #include <istream>
 #include <string>
 
+// use of illegal utf8 character for this purpose was added in a76be7ef1e921dabacd99f16ef440bf9673b8d98
+// added by the preprocessor to allow special handling for #line and #textdomain commands
+constexpr unsigned char INLINED_PREPROCESS_DIRECTIVE_CHAR = 254;
+
+// normal ascii is 0-127
+// extended ascii is from 128-255, none of which need any special handling
+constexpr int START_EXTENDED_ASCII = 128;
```

Comment:
127 would be the end of regular ascii though, not 128.

Reply:
By STL logic, it would be 128, because the end is excluded from the range.

Reply:
This doesn't have anything to do with the STL though.

Reply:
I'm saying it makes sense because it's consistent with how the standard C++ library does things. In a range, whether an iterator or numeric range (eg `std::string::substr`), the end of the range is always exclusive and the beginning of the range is always inclusive.

Reply:
Updated.